src/operations/indexes.ts

```diff
@@ -176,42 +181,12 @@ function makeIndexSpec(
 }
 
 /** @internal */
-export class IndexesOperation extends AbstractOperation<Document[]> {
-  override options: IndexInformationOptions;
-  collection: Collection;
-
-  constructor(collection: Collection, options: IndexInformationOptions) {
-    super(options);
-    this.options = options;
-    this.collection = collection;
-  }
-
-  override get commandName() {
-    return 'listIndexes' as const;
-  }
-
-  override async execute(_server: Server, session: ClientSession | undefined): Promise<Document[]> {
-    const coll = this.collection;
-    const options = this.options;
-
-    return indexInformation(coll.s.db, coll.collectionName, {
-      full: true,
-      ...options,
-      readPreference: this.readPreference,
-      session
-    });
-  }
-}
-
-/** @internal */
-export class CreateIndexesOperation<
-  T extends string | string[] = string[]
-> extends CommandOperation<T> {
+export class CreateIndexesOperation extends CommandOperation<string[]> {
   override options: CreateIndexesOptions;
   collectionName: string;
   indexes: ReadonlyArray<Omit<IndexDescription, 'key'> & { key: Map<string, IndexDirection> }>;
 
-  constructor(
+  private constructor(
```

Comment:
I'm happy to revert this change - but since I added static factories I decided it made sense to restrict usage of the constructor.